### DISCUSSION TITLE
Modbus: SAE liefert nur "0" an SHM

### DISCUSSION THREAD

**Discussion by runn8** [camueller/SmartApplianceEnabler#255] [Installation und Betrieb]
Hallo,
nach unzähligen Versuchen hoffe ich auf eine Idee in diesem Forum.

Ich nutze einen SDM630 3-Phasen Zähler mit ModBus an einen Raspi3 mittels mbusd. Die Abfrage mit mbpoll liefert auch schöne Werte, zB. hier:

> pi@raspberrypi:~ $ mbpoll -a 1 -r 0x34 -t 3:float -B -0 -1 localhost
> mbpoll 1.4-12 - FieldTalk(tm) Modbus(R) Master Simulator
> Copyright © 2015-2019 Pascal JEAN, https://github.com/epsilonrt/mbpoll
> This program comes with ABSOLUTELY NO WARRANTY.
> This is free software, and you are welcome to redistribute it
> under certain conditions; type 'mbpoll -w' for details.
> 
> Protocol configuration: Modbus TCP
> Slave configuration...: address = [1]
>                         start reference = 52, count = 1
> Communication.........: localhost, port 502, t/o 1.00 s, poll rate 1000 ms
> Data type.............: 32-bit float (big endian), input register table
> 
> -- Polling slave 1...
> [52]: 	1252.11

Dieselbe Registeradresse liefert im SAE nur den Wert "0" an den SHM. Auch habe ich bei den Einstellungen für Zähler nicht die Option "Byte-Reihenfolge" zur Verfügung, wo ich - lt. Anleitung - "Big Endian (Standard)" ausfüllen könnte. Keine Ahnung, ob das einen Einfluss hat.

<img width="662" alt="grafik" src="https://user-images.githubusercontent.com/103494067/164538835-4c84ad3d-9b35-4f14-9a7c-3dbc853b2145.png">

Freue mich jedenfalls über jeden Tipp!
Vielen Dank u. einen schönen Abend

**Comment by camueller** [comment]
Du hast bei `mbpoll` als Register `0x34` (also 34 im Hex-Format). Du musst das im SAE auch so angeben - ohne `0x` am Anfang ist das sonst 34 im Dezimalformat!

**Reply by runn8** [reply]
Danke für deine rasche Antwort!
Ja, das stimmt, das Foto war missverstädlich. Ich habe es allerdings auch mit 0x34 (nun wieder geändert) versucht oder mit dem Dezimalwert 52. Keine Änderung. Es wird nur "0" gesendet.
<img width="672" alt="grafik" src="https://user-images.githubusercontent.com/103494067/164555738-45449fb9-364a-4e93-9137-4ac4eba4c3c2.png">

Im SHM:
<img width="359" alt="grafik" src="https://user-images.githubusercontent.com/103494067/164556042-af8e9858-bad2-44d8-a488-e2b511033176.png">

und:
<img width="1070" alt="grafik" src="https://user-images.githubusercontent.com/103494067/164555931-e95d147c-2c45-4c28-b32e-be031460eaf8.png">

**Comment by camueller** [comment]
Ich würde nicht die Leistung abfragen, sondern den Zählerstand. Das ist genauer! Ich habe mal kurz in die Doku zum SDM630 geschaut, das sind dieselben Register wie bei meinem SDM72. Insofern kannst Du einfach meine Einstellungen verwenden:

![image](https://user-images.githubusercontent.com/4489628/164604406-987f61f6-afe7-4c21-b34a-b2fce8ef3c60.png)

**Reply by runn8** [reply]
Danke für deine Nachricht!
Der Ferhler lag an der fehlenden Byte-Reihenfolge "Big Endian (Standard)". Diese konnte ich nicht eintragen, da ich bei Datenwörter "1" anstatt "2" eingetragen hatte. Sagenhaft, dass ich das immer und immer wieder übersehen konnte.

Nun schaut es vorerst gut aus und es kommen Werte, auch für die Leistung.

DANKE für deine Unterstützung!

**Reply by runn8** [reply]
> Ich würde nicht die Leistung abfragen, sondern den Zählerstand. Das ist genauer! Ich habe mal kurz in die Doku zum SDM630 geschaut, das sind dieselben Register wie bei meinem SDM72. Insofern kannst Du einfach meine Einstellungen verwenden:
> 
> ![image](https://user-images.githubusercontent.com/4489628/164604406-987f61f6-afe7-4c21-b34a-b2fce8ef3c60.png)

Hallo,

ich habe nun einen zweiten Stromzähler ergänzt: SDM72D-M
Dieser liefert manuell auch sofort alle aktuellen Werte. Im SHM jedoch kommen die Werte stark verzögert an. Kann man das irgendwie beeinflussen? Liegt es an einer generellen Limitierung vom SHM?

Danke für jeden Tipp!
@camueller Vielen Dank für deine Lösung und Unterstützung. Hab einen kleinen Beitrag übermittelt. 

LG,
Mario

**Comment by camueller** [comment]
Was heißt denn "stark verzögert"? Der SAE fragt den Zählerstand alle 60s ab und berechnet daraus den Zählerstand. Der SHM fragt die Leistung beim SAE ebenfalls alle 60s ab, d.h. die angezeigte Leistung im Sunny Portal sollte max. 1-2 Minuten hinterherhängen.

Vielen Dank für den Beitrag via Paypal :-)

**Reply by runn8** [reply]
Ah, das erklärt alles. Fritz!Plug Zwischenstecker liefern den Stromverbrauch alle paar Sekunden. 1-2min kommt hin. Ich vermute, das Intervall lässt sich nicht verkürzen, oder?

**Reply by camueller** [reply]
Nein.
